### DISCUSSION TITLE
一个关于Rust的小问题 --- 怎么在main里面用dev-graph

### DISCUSSION THREAD

**Discussion by voidcenter** [z2o-k7e/halo2-step-by-step#10] [Q&A] [answered]
在 Fib 的教程里面，Yinghong花了很多时间让 dev-graph 工作 （从[这儿](https://youtu.be/YclxoJ069zg?si=9F5gG3QzouRHH4G3&t=2195)开始的约20分钟 ）。最后需要建一个 lib.rs 并把画图做成[一个feature](https://github.com/icemelon/halo2-examples/blob/master/src/fibonacci/example1.rs#L208)并[单独写出dependencies](https://github.com/icemelon/halo2-examples/blob/master/Cargo.toml#L13)才运行通。

我想问一下这里有可能在 main 函数里运行 dev-graph 吗？ 我试了一下不知道怎么写dependency。具体地说不知道怎么把这种[写成feature的代码](https://github.com/zcash/halo2/blob/main/halo2_proofs/src/dev.rs#L34-L39)在 [dependencies] 里面写出来让整个项目都能用。如果不能在 main 里面画图，必须要起一个 lib + feature 来画图感觉还挺别扭的。

**Comment by Demian101** [comment, marked as answer]
这是个 Rust 问题，感觉  Chatgpt 答得还还不错：

https://chat.openai.com/c/12156005-4b55-4cd6-a421-38e3b8b6f669
<img width="561" alt="image" src="https://github.com/zkp-co-learning/halo2-step-by-step/assets/33189338/543f1e35-df34-468b-8453-aa8819fc6d3c">

**Reply by voidcenter** [reply]
确实科技改变生活啊 :grinning:
